.github/workflows/test.yml

```diff
@@ -0,0 +1,183 @@
+        run: pip install build
+      - name: Build the package
+        run: python -m build
+      - name: Upload the build artifact
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4
+        with:
+          name: anaconda-cli-base-wheel-${{ github.sha }}
+          path: dist/*
+          if-no-files-found: error
+          retention-days: 7
+
+  # This check job runs to ensure all tests and builds  have passed, such that we can use it as a "wildcard"
+  # for branch protection to ensure all tests pass before a PR can be merged.
+  check:
+    name: Check all tests passed
+    if: always()
+    needs: [test, build-conda-package, build-wheel]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Decide whether all required jobs succeeded or failed
+      uses: re-actors/alls-green@release/v1
+      with:
+        jobs: ${{ toJSON(needs) }}
+
+  publish-conda-pkg-to-anaconda-dot-org:
+    name: Publish conda package to Anaconda.org
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'  # Only run on push to main branch
+    needs: [check]
+    steps:
+    - name: Retrieve the source code
+      uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4
+      with:
+        fetch-depth: 0
+    - name: Create build environment
+      run: |
+        source $CONDA/bin/activate
+        conda create -n build --file ./etc/build.linux-64.lock
+    - name: Download the build artifacts
+      uses: actions/download-artifact@9c19ed7fe5d278cd354c7dfd5d3b88589c7e2395 # v4
+      with:
+        name: anaconda-cli-base-conda-${{ github.sha }}
+        path: ~/anaconda-cli-base-conda-bld
+    - name: publish
+      env:
+        TOKEN: ${{ secrets.ANACONDA_ORG_UPLOAD_TOKEN }}
+      run: |
+        source $CONDA/bin/activate && conda activate build
+        [[ "$GITHUB_REF" =~ ^refs/tags/v ]] || export LABEL="--label dev"
+        anaconda --verbose \
+          --token $TOKEN \
+          upload \
+          --user anaconda-cloud \
+          $LABEL \
+          --force \
+          --private \
+          ~/anaconda-cli-base-conda-bld/noarch/anaconda-cli-base-*
+
+  publish-wheel-to-anaconda-dot-org:
+    name: Publish wheel to Anaconda.org
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'  # Only run on push to main branch
+    needs: [check]
+    steps:
+    - name: Retrieve the source code
+      uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4
+      with:
+        fetch-depth: 0
+    - name: Download the build artifacts
+      uses: actions/download-artifact@9c19ed7fe5d278cd354c7dfd5d3b88589c7e2395 # v4
+      with:
+        name: anaconda-cli-base-wheel-${{ github.sha }}
+        path: ~/dist
+    - name: Upload to anaconda.org
+      env:
+        TOKEN: ${{ secrets.ANACONDA_ORG_UPLOAD_TOKEN }}
+        GITHUB_REF: ${{ github.ref }}
+      run: |
+        source $CONDA/bin/activate
+        conda install -y anaconda-client
```

Comment:
Is there a reason to do this differently here than re-using the build environment like above?